litmus-portal/frontend/src/views/ChaosWorkflows/BrowseAnalytics/WorkflowRunDetailsTable/index.tsx

```diff
@@ -0,0 +1,396 @@
+      fetchPolicy: 'cache-and-network',
+    }
+  );
+
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const emptyRows =
+    rowsPerPage -
+    Math.min(rowsPerPage, displayData.length - page * rowsPerPage);
+
+  const getTests = (searchingData: workFlowTests[]) => {
+    const uniqueList: string[] = [];
+    searchingData.forEach((data) => {
+      if (!uniqueList.includes(data.test_name)) {
+        uniqueList.push(data.test_name);
+      }
+    });
+    setTests(uniqueList);
+  };
+
+  const getTestResults = (searchingData: workFlowTests[]) => {
+    const uniqueList: string[] = [];
+    searchingData.forEach((data) => {
+      if (!uniqueList.includes(data.test_result)) {
+        uniqueList.push(data.test_result);
+      }
+    });
+    setTestResults(uniqueList);
+  };
+
+  useEffect(() => {
+    const processedWorkflowRunDetails: workFlowTests[] = [];
+    let experimentWeights: Weights[] = [];
+    const points: number[] = [];
+    const weights: number[] = [];
+
+    data?.getScheduledWorkflows.forEach((data) => {
+      try {
+        if (data.workflow_id === workflowID) {
+          experimentWeights = data.weightages;
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    });
+
+    workflowRunDetails.forEach((detail) => {
+      try {
+        experimentWeights.forEach((mappedWeight) => {
+          try {
+            if (detail.test_name === mappedWeight.experiment_name) {
+              processedWorkflowRunDetails.push({
+                test_id: detail.test_id,
+                test_name: mappedWeight.experiment_name,
+                test_result: detail.test_result,
+                weight: mappedWeight.weightage,
+                resulting_points:
+                  (detail.test_result === 'Pass' ? 1 : 0) *
+                  mappedWeight.weightage,
+                last_run: detail.last_run,
+              });
+              points.push(
+                (detail.test_result === 'Pass' ? 1 : 0) * mappedWeight.weightage
+              );
+              weights.push(mappedWeight.weightage);
+            }
+          } catch (error) {
+            console.error(error);
+          }
+        });
+      } catch (error) {
+        console.error(error);
+      }
+    });
+
+    setMainData(processedWorkflowRunDetails);
+    setDisplayData(processedWorkflowRunDetails);
+    setResilienceScore(
+      parseFloat(
+        (
+          (points.reduce((a, b) => a + b, 0) /
+            weights.reduce((a, b) => a + b, 0)) *
+          100
+        ).toFixed(2)
+      )
+    );
+    getTests(processedWorkflowRunDetails);
+    getTestResults(processedWorkflowRunDetails);
+    setReload(true);
+  }, [data, loading, workflowID, reload]);
+
+  useEffect(() => {
```

Comment:
filter is a state variable it'll automatically recalculate this. no need for useEffect here

Reply:
I am updating the selection options dynamically upon changing he filter @arkajyotiMukherjee , if i don't use useEffect() how should i call the functions to update the drop down lists for test and tet results?

Reply:
this is basic concept of state in react...we have done the same in browse workflow
filter is a state var. hence when it changes react will rerender the component where it is used...no need of useEffect

Reply:
That is good but I am calling 3 functions PTAL, how do I call them without use effect?